cmd/ndm_daemonset/probe/eventhandler.go

```diff
@@ -110,3 +153,254 @@ func (pe *ProbeEvent) initOrErrorEvent() {
+					// 1. deactivate parent
+					// 2. create resource for partition
+
+					pe.Controller.DeactivateBlockDevice(*parentBDAPI)
+					deviceInfo := pe.Controller.NewDeviceInfoFromBlockDevice(&bd)
+					existingBlockDeviceResource := pe.Controller.GetExistingBlockDeviceResource(bdAPIList, deviceInfo.UUID)
+					err := pe.Controller.PushBlockDeviceResource(existingBlockDeviceResource, deviceInfo)
+					if err != nil {
+						klog.Error(err)
+						return err
+					}
+					return nil
+				}
+
+			}
+
+			if bd.DeviceAttributes.DeviceType != libudevwrapper.UDEV_PARTITION &&
+				len(bd.DependentDevices.Partitions) > 0 {
+				klog.V(4).Infof("device: %s has partitions: %+v", bd.DevPath, bd.DependentDevices.Partitions)
+				return nil
+			}
+
+			if len(bd.DependentDevices.Holders) > 0 {
+				klog.V(4).Infof("device: %s has holders: %+v", bd.DevPath, bd.DependentDevices.Holders)
+				return nil
+			}
+
+			klog.V(4).Infof("creating block device resource for device: %s with uuid: %s", bd.DevPath, bd.UUID)
+			deviceInfo := pe.Controller.NewDeviceInfoFromBlockDevice(&bd)
+			existingBlockDeviceResource := pe.Controller.GetExistingBlockDeviceResource(bdAPIList, deviceInfo.UUID)
+			err := pe.Controller.PushBlockDeviceResource(existingBlockDeviceResource, deviceInfo)
+			if err != nil {
+				klog.Errorf("could not create block device resource, %+v", err)
+				return err
+			}
+			return nil
+		}
+
+		if err != nil {
+			klog.Errorf("querying etcd failed: %+v", err)
+			return err
+		}
+
+		if bdAPI.Status.ClaimState != apis.BlockDeviceUnclaimed {
+			klog.V(4).Infof("device: %s is in use. update the details of the blockdevice", bd.DevPath)
+			deviceInfo := pe.Controller.NewDeviceInfoFromBlockDevice(&bd)
+			err = pe.Controller.PushBlockDeviceResource(bdAPI, deviceInfo)
+			if err != nil {
+				klog.Errorf("updating block device resource failed: %+v", err)
+				return err
+			}
+			return nil
+		}
+
+		klog.V(4).Infof("creating resource for device: %s with uuid: %s", bd.DevPath, bd.UUID)
+		deviceInfo := pe.Controller.NewDeviceInfoFromBlockDevice(&bd)
+		existingBlockDeviceResource := pe.Controller.GetExistingBlockDeviceResource(bdAPIList, deviceInfo.UUID)
+		err = pe.Controller.PushBlockDeviceResource(existingBlockDeviceResource, deviceInfo)
+		if err != nil {
+			klog.Errorf("creation of resource failed: %+v", err)
+			return err
+		}
+		return nil
+	} else {
+		klog.V(4).Infof("device: %s cannot be uniquely identified", bd.DevPath)
+		if len(bd.DependentDevices.Partitions) > 0 ||
+			len(bd.DependentDevices.Holders) > 0 {
+			klog.V(4).Infof("device: %s has holders/partitions. %+v", bd.DevPath, bd.DependentDevices)
+		} else {
+			klog.Infof("starting to create partition on device: %s", bd.DevPath)
+			d := partition.Disk{
+				DevPath:          bd.DevPath,
+				DiskSize:         bd.Capacity.Storage,
+				LogicalBlockSize: uint64(bd.DeviceAttributes.LogicalBlockSize),
+			}
+			if err := d.CreateSinglePartition(); err != nil {
```

Comment:
Yes. A single GPT partition that spans the entire disk will be created. When the partition table is written it will again trigger a udev add event for the partition. During processing this event, a UUID will be generated for the partition and BlockDevice resource will be created.